### PR TITLE
Refactor navigation UI

### PR DESCRIPTION
**Pull request by seanyesmunt**
#### Notes
- A lot of the changes come from having to create several components
- ClaimListDiscover is a little simpler, and now requires more props to be passed in from wherever it's being used
- Will revert all the link changes back to `/` and get rid of the redirect on the server when we have an actual home page
- I didn't add the persistent filters on these new pages. That will take a little more work and I didn't want to make this PR even bigger.

### CODE REVIEW

File: ui/page/channelsFollowing/view.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    if (!hasSubsribedChannels) {
+      doFetchRecommendedSubscriptions();
+    }
+  }, [doFetchRecommendedSubscriptions, hasSubsribedChannels]);
+
+  return (
+    <Page>
+      {showTab ? (
+        <ClaimList
+          headerLabel={__('Find Channels To Follow')}
```
Comment (kauffj) [inline ui/page/channelsFollowing/view.jsx]: Find Channels to Follow

File: ui/page/channelsFollowingManage/view.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    let url = `/$/${PAGES.CHANNELS_FOLLOWING_MANAGE}`;
+    if (!viewingSuggestedSubs) {
+      url += '?view=discover';
+    }
+
+    history.push(url);
+  }
+
+  useEffect(() => {
+    doFetchRecommendedSubscriptions();
```
Comment (kauffj) [inline ui/page/channelsFollowingManage/view.jsx]: If you want to kill this and replace it with `claim_search` for top channels by bid or whatever I would be all for it.

File: lbrytv/middleware/redirect.js [hunk 3]
```diff
@@ -13,6 +13,11 @@ async function redirectMiddleware(ctx, next) {
     return;
   }
 
+  if (path === '/') {
```
Comment (kauffj) [inline lbrytv/middleware/redirect.js]: conditional by user auth state until new homepage is done? this won't make much sense for unauth users

File: ui/analytics.js [hunk 4]
```diff
@@ -46,7 +46,7 @@ type LogPublishParams = {
 let analyticsEnabled: boolean = true;
```
[the remark this answers is not in the document]
Comment (seanyesmunt) [reply ui/analytics.js]: Ah that's a good idea.

File: lbrytv/middleware/redirect.js (hunk 3 above)
Comment (seanyesmunt) [reply lbrytv/middleware/redirect.js]: That's harder to do unless we call internal-apis with the auth token. I figured we could still keep the url (even if it's a little confusing) but still show the signed out view

**Pull request merged**